### PR TITLE
DAOS-5018 Hadoop compatible exception (#2800)

### PR DESCRIPTION
**Pull request by zjf2012**
In current impl of Hadoop DAOS, some exceptions are not compatible with Hadoop protocol. In this bug fixing, we'll throw exceptions defined by Hadoop protocol as much as possible.

Signed-off-by: jiafu zhang <jiafu.zhang@intel.com>

### CODE REVIEW

**Comment by zjf2012** [comment]
@johannlombardi , please help review this PR.

File: src/client/java/hadoop-daos/src/test/java/io/daos/fs/hadoop/DaosFileSystemContractIT.java [hunk 1]
```diff
@@ -207,15 +120,17 @@ public void testGetFileStatusFileAndDirectory() throws Exception {
...
   public void testMkdirsForExistingFile() throws Exception {
+    try{
     Path testFile = this.path("/test/hadoop/file");
     assertFalse(this.fs.exists(testFile));
-    this.fs.mkdirs(new Path("/test/hadoop"));
     this.createFile(testFile);
     assertTrue(this.fs.exists(testFile));
     this.fs.mkdirs(testFile);
+//    fail("/test/hadoop/file is a file");
+    }catch (FileAlreadyExistsException e){
```
Comment (johannlombardi) [inline src/client/java/hadoop-daos/src/test/java/io/daos/fs/hadoop/DaosFileSystemContractIT.java:130]: please add space before/after {}

Comment (zjf2012) [reply src/client/java/hadoop-daos/src/test/java/io/daos/fs/hadoop/DaosFileSystemContractIT.java:130]: we usually don't check the styling in test code. But as you pointed out, I'll check them all. thanks.

Comment (zjf2012) [reply src/client/java/hadoop-daos/src/test/java/io/daos/fs/hadoop/DaosFileSystemContractIT.java]: done

**Comment by zjf2012** [comment]
fixed styling issue. @johannlombardi , please check.

**Comment by zjf2012** [comment]
@johannlombardi , do you have more comments or need someone else to review it? If not, please help merge it so that I can submit DAOS UNS wrapper PR.
thanks.

**Pull request merged**